BurgerQueen/src/app/@core/authentication/services/auth.service.spec.ts

```diff
@@ -23,75 +21,69 @@ class MockLocalStorageService {
+  };
 
   beforeEach(() => {
-    httpClientSpy = jasmine.createSpyObj('HttpClient', ['post']);
-
     TestBed.configureTestingModule({
       imports: [HttpClientTestingModule],
-      providers: [
-        AuthService,
-        { provide: MockLocalStorageService, useClass: MockLocalStorageService },
-        { provide: HttpClient, useValue: httpClientSpy },
-      ],    });
-    
+      providers: [AuthService, MockLocalStorageService],
+    });
+
     authService = TestBed.inject(AuthService);
     httpTestingController = TestBed.inject(HttpTestingController);
     localStorageService = TestBed.inject(MockLocalStorageService);
   });
 
   describe('Login', () => {
-    beforeEach(()=> {
-      httpClientSpy.post.and.returnValue(of(mockResponse));
+    it('should make a POST request to the login endpoint with correct credentials ', () => {
       authService.login(mockCredentials);
-    });
+      const req = httpTestingController.expectOne(
+        `${authService['apiUrl']}/login`
+      );
 
-    it('should send a POST request when calling login', () => {
-      expect(httpClientSpy.post).toHaveBeenCalledWith(`${environment.apiUrl}/login`, mockCredentials);
-      
-      authService.loginResponse$.subscribe((response) => {
-        expect(response.data).toEqual(mockResponse);
-      });
+      expect(req.request.method).toEqual('POST');
+      expect(req.request.body).toEqual(mockCredentials);
+
+      req.flush(mockResponse);
     });
-    
-    it('call the method login must set isloading a true',
-    (done: DoneFn) => {
-      httpClientSpy.post.and.returnValue(of(mockResponse));   
-      authService.loginResponse$.pipe(first()).
-      subscribe(response => {
-        expect(response.isLoading).toBeTrue();
-        done();
-      });
+
+    it('should handle login errors', () => {
+      const errorMessage = 'Login failed';
       authService.login(mockCredentials);
-    }
-    )
-
-    it('call the method login if the response is valid must set system user',
-      (done: DoneFn) => {
-        httpClientSpy.post.and.returnValue(of(mockResponse));   
-        authService.login(mockCredentials);
-        authService.systemUser$.
-        subscribe(response => {
-          expect(response).toEqual({id: '1', accessToken: 'mockAccessToken', role: 'user', email: 'test@test.com'});
-          done();
+      httpTestingController.expectOne(`${authService['apiUrl']}/login`).error(
+        new ErrorEvent('Network error', {
+          message: errorMessage,
+        })
+      );
+
+      authService.systemUser$.subscribe((user) => {
+        expect(user).toEqual({
+          id: '',
+          accessToken: '',
+          role: '',
+          email: '',
         });
-      }
-    )
-  })
-  
+
+        expect(localStorageService.getItem('accessToken')).toBe(null);
+        expect(localStorageService.getItem('role')).toBe(null);
+        expect(localStorageService.getItem('idUser')).toBe(null);
+      });
+    });
```

Comment:
Esta prueba dice que valida que se manajen los errores pero el codigo nos dice 

https://github.com/Luandrada/DEV006-burger-queen-api-client/blob/a71ad3e1f37045f4e771b1b7d821f318a24b14fe/BurgerQueen/src/app/%40core/authentication/services/auth.service.ts#L28

que si no hay un data (como seria el caso de un error) no el next del systemUser no se ejecuta por lo que no tendria mucho sentido la prueba 

si queremos asegurar que se manejo el error lo que deberiamos ver el es valor de la propiedad error de loginResponse tiene lo que esperamos 

porque esto es lo que realmente usamos en el exterior 

https://github.com/Luandrada/DEV006-burger-queen-api-client/blob/a71ad3e1f37045f4e771b1b7d821f318a24b14fe/BurgerQueen/src/app/%40core/authentication/sign-in/sign-in.component.ts#L29